Bottom: 01a7cd717b66fb736f98bedbaec389f2a5137001
Top:    f6ee6be12f8331629ab0d0f559cf2c7ac55ceb16
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-07-14 14:02:38 +0800

Refresh of add-a-vq-meta-data-iotlb

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 28969dc..79001be 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -720,10 +720,14 @@ static void __user *vhost_vq_iotlb_fetch(struct vhost_virtqueue *vq,
 	case VHOST_ADDR_AVAIL:
 	case VHOST_ADDR_USED:
 		node = vq->vq_iotlb[type];
+		break;
 	default:
 		BUG();
 	}
 
+	if (!node)
+		return NULL;
+
 	BUG_ON(addr < node->start || (u64)addr > node->start + node->size);
 
 	if ((u64)addr + size >= node->start + node->size)
@@ -1166,6 +1170,7 @@ static void vhost_vq_iotlb_update(struct vhost_virtqueue *vq,
 	case VHOST_ADDR_AVAIL:
 	case VHOST_ADDR_USED:
 		vq->vq_iotlb[type] = node;
+		break;
 	default:
 		BUG();
 	}
